auth/src/main/java/com/firebase/ui/auth/util/SmartLock.java

```diff
@@ -201,7 +205,11 @@ public void onActivityResult(int requestCode, int resultCode, Intent data) {
     }
 
     private void finish(int resultCode) {
-        mActivity.finish(RESULT_OK, mActivity.getIntent());
+        mActivity.finish(resultCode, mActivity.getIntent());
+    }
+
+    private void finish(int resultCode, Intent data) {
+        mActivity.finish(resultCode, data);
```

Comment:
You'll need to do some more refactoring for this to work. The reason I chose not to pass down the `resultCode` is because `SmartLock` used to be an Activity called by other activities (such as `RegisterEmailActivity`). Those activities didn't care about what the result code was and always returned `Activity.RESULT_OK`. Say you start the email flow through `AuthMethodPickerActivity` and end up at `RegisterEmailActivity` which calls this smart lock code. If saving the credential fails,`RegisterEmailActivity` will be finished with `Activity.RESULT_CANCELED`. However, the user is signed in and [this line in `AuthMethodPickerActivity`](https://github.com/amandle/FirebaseUI-Android/blob/470147405e20aa52599544ff7000ef04ddc928e4/auth/src/main/java/com/firebase/ui/auth/ui/idp/AuthMethodPickerActivity.java#L146) will prevent the activity from finishing even though the user is singed in. I don't think this is expected behavior.